js/models/messages.js

```diff
@@ -518,7 +518,15 @@
                                     catch (e) {
                                         handleError(e);
                                     }
-                                }, handleError);
+                                }, function(error) {
+                                    console.log('handleDataMessage: Message', message.idForLogging(), 'was deleted');
+
+                                    if (confirm) {
```

Comment:
To make sure that I'm understanding, this change means when we fail to fetch we now delete the message from the unprocessed message cache?

Should confirm ever be null?